library/core/src/intrinsics.rs

```diff
@@ -2802,7 +2802,6 @@ where
 /// In other words, the following code has *Undefined Behavior*:
```

Comment:
Should this intrinsic be unsafe?

Reply:
```rust
pub const fn is_val_statically_known<T: Copy>(_arg: T) -> bool {
```

it doesn't appear to be...?

Reply:
OK, discussed a bit in the meeting, I see that the unsafe thing is the use of *unreachable*, but it still feels like the user has a kind of obligation to use the result of this value to produce an "equivalent" results (i.e., it's intended for "performance specialization"). I suppose that we no longer guarantee deterministic results so maybe that's par for the course and not *unsafe*

Reply:
Yeah, we made `const_eval_select` safe with the same reasoning -- it's not UB-inducing, just undesired.